pkg/kubelet/kuberuntime/kuberuntime_manager.go

```diff
@@ -1099,7 +1099,13 @@ func (m *kubeGenericRuntimeManager) SyncPod(ctx context.Context, pod *v1.Pod, po
 
 		// Prepare resources allocated by the Dynammic Resource Allocation feature for the pod
 		if utilfeature.DefaultFeatureGate.Enabled(features.DynamicResourceAllocation) {
-			if m.runtimeHelper.PrepareDynamicResources(pod) != nil {
+			if err := m.runtimeHelper.PrepareDynamicResources(pod); err != nil {
+				ref, referr := ref.GetReference(legacyscheme.Scheme, pod)
+				if referr != nil {
+					klog.ErrorS(referr, "Couldn't make a ref to pod", "pod", klog.KObj(pod))
+				}
+				m.recorder.Eventf(ref, v1.EventTypeWarning, events.FailedPrepareDynamicResources, "Failed to prepare dynamic resources: %v", err)
+				klog.ErrorS(err, "Failed to prepare dynamic resources", "pod", klog.KObj(pod))
```

Comment:
I believe you need the additional return statement otherwise you end up accessing a nil ref later on:
```suggestion
			if err := m.runtimeHelper.PrepareDynamicResources(pod); err != nil {
				ref, referr := ref.GetReference(legacyscheme.Scheme, pod)
				if referr != nil {
					klog.ErrorS(referr, "Couldn't make a ref to pod", "pod", klog.KObj(pod))
					return
				}
				m.recorder.Eventf(ref, v1.EventTypeWarning, events.FailedPrepareDynamicResources, "Failed to prepare dynamic resources: %v", err)
				klog.ErrorS(err, "Failed to prepare dynamic resources", "pod", klog.KObj(pod))
```

Reply:
+1 and maybe add the `err` to the log in this case so the `err` will be at least in logs

Reply:
looking at the code around - this is actually a pattern. @bart0sh can you check what will be the behavior in case of `nil` and maybe more instance of this behavior needs to be fixed in this file

Reply:
@klueska done, thanks for pointing out.

@SergeyKanzhelev considering that the code freeze is coming, I'd like to come up with another PR covering similar cases in the code around, so this one gets a chance to be accepted before the freeze. Makes sense?